post-scan-actions/azure-python-promote-or-quarantine/template.json

```diff
@@ -0,0 +1,283 @@
+            },
+            "properties": {
+                "Application_Type": "web",
+                "ApplicationId": "[variables('applicationInsightsName')]"
+            }
+        },
+        {
+            "apiVersion": "2019-05-01",
+            "name": "scanResultSubscription",
+            "type": "Microsoft.Resources/deployments",
+            "resourceGroup": "[variables('scanResultTopicResourceGroupName')]",
+            "subscriptionId": "[variables('scanResultTopicResourceSubscriptionID')]",
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {},
+                    "variables": {},
+                    "resources": [
+                        {
+                            "type": "Microsoft.ServiceBus/namespaces/topics/subscriptions",
+                            "apiVersion": "[variables('scanResultTopicAPIVersion')]",
+                            "name": "[format('{0}/{1}/{2}', variables('scanResultTopicNamespace'), variables('scanResultTopicName'), variables('scanResultTopicSubscriptionName'))]",
+                            "properties": {
+                                "defaultMessageTimeToLive": "PT1H",
+                                "maxDeliveryCount": 3
+                            }
+                        }
+                    ]
+                },
+                "parameters": {}
+            }
+        },
+        {
+            "apiVersion": "2019-05-01",
+            "name": "scanningStorageBlobDataOwnerRoleAssignment",
+            "type": "Microsoft.Resources/deployments",
+            "resourceGroup": "[variables('scanningStorageResourceGroupName')]",
+            "subscriptionId": "[variables('scanningStorageSubscriptionID')]",
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {},
+                    "variables": {},
+                    "resources": [
+                        {
+                            "type": "Microsoft.Authorization/roleAssignments",
+                            "apiVersion": "2018-09-01-preview",
+                            "name": "[variables('scanningStorageBlobDataOwnerRoleAssignmentUID')]",
+                            "scope": "[concat('Microsoft.Storage/storageAccounts/', variables('scanningStorageAccountName'))]",
+                            "properties": {
+                                "roleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', variables('blobDataOwnerRoleID'))]",
```

Comment:
Does the function require all the permissions of protecting blob or it just need to read and delete?

Reply:
It requires read, get delegation key and delete permissions. I'll update to use a custom role.